lib/ocpp/v201/charge_point.cpp

```diff
@@ -416,6 +423,62 @@ void ChargePoint::handle_scheduled_change_availability_requests(const int32_t ev
     }
 }
 
+bool ChargePoint::is_transaction_ongoing(const CiString<36>& transaction_id) {
+    for (auto const& [evse_id, evse] : evses) {
+        if (evse->has_active_transaction()) {
+            if (transaction_id == evse->get_transaction()->get_transaction().transactionId) {
+                return true;
+            }
+        }
+    }
+
+    return false;
+}
+
+bool ChargePoint::is_evse_connector_reserved_not_available(const std::unique_ptr<Evse>& evse, const IdToken& id_token,
+                                                           const std::optional<IdToken>& group_id_token) const {
+    const int32_t connectors = evse->get_number_of_connectors();
+    for (int32_t i = 0; i < connectors; ++i) {
+        const ConnectorStatusEnum& status = evse->get_state(i);
+        if (status == ConnectorStatusEnum::Reserved) {
+            const CiString<36>& transaction_id_token = evse->get_transaction()->id_token.idToken;
```

Comment:
`evse->get_transaction()` can be nullptr or almost certainly is a nullptr in this if case.

If a transaction is active the status should be 'Occupied' and not 'Reserved'. 

I think we need some completely different handling for this. There could be a callback that asks the Auth mechanism (outside) of libocpp if this remote request / auth request should accepted or rejected.